src/close.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{transaction::CloseTransaction, Channel};
+use bitcoin::{Address, Transaction};
+use ecdsa_fun::Signature;
+
+pub struct State0(Channel);
+pub struct State1 {
+    channel: Channel,
+    final_address_self: Address,
+    final_address_other: Address,
+}
+pub struct FinalState(Transaction);
+
+impl FinalState {
+    pub fn into_transaction(self) -> Transaction {
+        self.0
+    }
+}
+
+impl From<Channel> for State0 {
+    fn from(channel: Channel) -> Self {
+        Self(channel)
+    }
+}
+
+pub struct Message0 {
+    final_address: Address,
+}
+
+#[derive(Debug)]
+pub struct Message1 {
+    close_transaction: CloseTransaction,
+    sig_close_transaction: Signature,
+}
+
+impl State0 {
+    pub fn compose(&self, final_address: Address) -> Message0 {
+        Message0 { final_address }
+    }
+
+    pub fn interpret(
+        self,
+        final_address_self: Address,
+        Message0 {
+            final_address: final_address_other,
+        }: Message0,
+    ) -> State1 {
+        State1 {
+            channel: self.0,
+            final_address_self,
+            final_address_other,
+        }
+    }
+}
+
+impl State1 {
+    pub fn compose(&self) -> anyhow::Result<Message1> {
+        let (amount_a, ownership_a) = self.channel.current_state.signed_TX_s.outputs().a;
```

Comment:
:upside_down_face: To be consistent we could use `X_a` and `X_b` instead of `ownership_a` and `ownership_b` respectively.